examples/custom-listeners/src/test/java/org/apache/james/examples/custom/listeners/PositionCustomFlagOnBigMessagesTest.java

```diff
@@ -0,0 +1,201 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.examples.custom.listeners;
+
+import static org.apache.james.examples.custom.listeners.PositionCustomFlagOnBigMessages.BIG_MESSAGE;
+import static org.apache.james.examples.custom.listeners.PositionCustomFlagOnBigMessages.ONE_MB;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Stream;
+
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.DefaultMailboxes;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MailboxSessionUtil;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxManager;
+import org.apache.james.mailbox.inmemory.manager.InMemoryIntegrationResources;
+import org.apache.james.mailbox.model.ComposedMessageId;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mailbox.store.event.EventFactory;
+import org.apache.james.mime4j.dom.Message;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Streams;
+
+class PositionCustomFlagOnBigMessagesTest {
+
+    private static final String USER = "user";
+    private static final Event.EventId RANDOM_EVENT_ID = Event.EventId.random();
+    private static final MailboxPath INBOX_PATH = MailboxPath.forUser(USER, DefaultMailboxes.INBOX);
+
+    private PositionCustomFlagOnBigMessages testee;
+    private MessageManager inboxMessageManager;
+    private MailboxId inboxId;
+    private MailboxSession mailboxSession;
+    private InMemoryMailboxManager mailboxManager;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        InMemoryIntegrationResources resources = InMemoryIntegrationResources.defaultResources();
+        mailboxManager = resources.getMailboxManager();
+        mailboxSession = MailboxSessionUtil.create(USER);
+        inboxId = mailboxManager.createMailbox(INBOX_PATH, mailboxSession).get();
+        inboxMessageManager = mailboxManager.getMailbox(inboxId, mailboxSession);
+
+        testee = new PositionCustomFlagOnBigMessages(mailboxManager);
+
+        resources.getEventBus().register(testee);
+    }
+
+    @Test
+    void listeningEventShouldNotAddFlagWhenSmallMessages() throws Exception {
```

Comment:
why "listeningEvent" ? I would not put any subject here because a mailet has a single purpose